mycroft/stt/__init__.py

```diff
@@ -201,6 +201,29 @@ def execute(self, audio, language=None):
         return self.recognizer.recognize_houndify(audio, self.id, self.key)
 
 
+class GoVivaceSTT(TokenSTT):
+    def __init__(self):
+        super(GoVivaceSTT, self).__init__()
+        self.default_uri = "https://services.govivace.com:49149/telephony"
+
+        if not self.lang.startswith("en") and not self.lang.startswith("es"):
+            LOG.error("GoVivace STT only supports english and spanish")
+            raise NotImplementedError
+
+    def execute(self, audio, language=None):
+        url = self.config.get("uri", self.default_uri) + "?key=" + self.token + \
+              "&action=find&format=8K_PCM16&validation_string="
+        response = put(url,
+                       data=audio.get_wav_data())
+        return self.get_response(response)
+
+    def get_response(self, response):
+        try:
+            return response.json()["result"]["hypotheses"][0]["transcript"]
+        except:
+            return None
```

Comment:
This except catch all is not helpful, it eats any indication of what is wrong and returns `None` which raises an error in the listener instead.

Reply:
kaldi does it, aren't all STT engines supposed to return None if speech was not recognized?